HTML5SDK/wwtlib/Layers/SpreadSheetLayer.cs

```diff
@@ -2031,6 +2030,108 @@ public int SizeColumn
                 }
             }
         }
+
+        // The following attributes control whether the point sizes should be normalized before
+        // being used. When NormalizeSize is true, the point sizes are scaled using
+        //
+        // new_size = (size - NormalizeSizeMin) / (NormalizeSizeMax - NormalizeSizeMin)
+        //
+        // The NormalizeSizeClip attribute can be used to determine whether the sizes should
+        // be clipped to the range [0:1]. At this time, normalization is only applied if
+        // PointScaleTypes is Linear or Power.
+
+        protected bool normalizeSize = false;
+
+        public bool NormalizeSize
+        {
+            get { return normalizeSize; }
+            set
+            {
+                if (normalizeSize != value)
+                {
+                    version++;
+                    normalizeSize = value;
+                }
+            }
+        }
+
+        protected bool normalizeSizeClip = false;
+
+        public bool NormalizeSizeClip
+        {
+            get { return normalizeSizeClip; }
+            set
+            {
+                if (normalizeSizeClip != value)
+                {
+                    version++;
+                    normalizeSizeClip = value;
+                }
+            }
+        }
+
+        protected float normalizeSizeMin = 0;
+
+        public float NormalizeSizeMin
+        {
+            get { return normalizeSizeMin; }
+            set
+            {
+                if (normalizeSizeMin != value)
+                {
+                    version++;
+                    normalizeSizeMin = value;
+                }
+            }
+        }
+
+        protected float normalizeSizeMax = 1;
+
+        public float NormalizeSizeMax
+        {
+            get { return normalizeSizeMax; }
+            set
+            {
+                if (normalizeSizeMax != value)
+                {
+                    version++;
+                    normalizeSizeMax = value;
+                }
+            }
+        }
+
+        public float NormalizePointSize(float value)
+        {
+
+            float new_value;
+
+            if(NormalizeSize)
+            {
+                new_value = (value - NormalizeSizeMin) / (NormalizeSizeMax - NormalizeSizeMin);
+
+                if (NormalizeSizeClip)
+                {
+                    if (new_value < 0)
+                    {
+                        new_value = 0;
+                    }
+                    else if (new_value > 1)
+                    {
+                        new_value = 1;
+                    }
+                }
+
+            }
+            else
+            {
+                new_value = value;
+            }
```

Comment:
Style bikeshedding: I think it's good to "resolve" `if` statements briefly, if possible, so that the reader doesn't have to remember what condition is being considered. In this particular case, what I'm getting at is:

```
if(!NormalizeSize)
    return value;

float new_value = (value - NormalizeSizeMin)/ ....;
if (!NormalizeSizeClip)
    return new value;

if (new_value < 0)
...
return new_value;
```

This reduces rightward drift and, I'd like to think, cognitive load on the reader, since they can "retire" different conditions in the terse `if` clauses.

Reply:
Should be fixed in https://github.com/WorldWideTelescope/wwt-web-client/pull/237/commits/7d7a657fa2474fe4fed69205b8b6894d13311268 - let me know if that's not what you had in mind.